src/dvc_render/vega_templates.py

```diff
@@ -508,16 +501,35 @@ class SmoothLinearTemplate(Template):
                     },
                     "color": {"field": "rev", "type": "nominal"},
                 },
-                "layer": [
+                "transform": [
                     {
-                        "mark": {
-                            "type": "line",
-                            "point": True,
-                            "tooltip": {"content": "data"},
-                        }
-                    }
+                        "loess": Template.anchor("y"),
+                        "on": Template.anchor("x"),
+                        "groupby": ["rev", "filename"],
```

Comment:
Q: do we still need `filename` here though?

Reply:
I think so, and we should add 'field` I think? It depends on how VS Code handles it, but I assume we want to groupby each line on the plot, which is equivalent to each unique revison-filename-field combo.

Reply:
Added `field`.

Reply:
Added `filename::field` for now per https://github.com/iterative/vscode-dvc/issues/3130.